functions.cmake

```diff
@@ -24,15 +24,40 @@ function(get_compile_definitions DefinitionName)
     set(${DefinitionName} ${DEFINITIONS} PARENT_SCOPE)
 endfunction(get_compile_definitions)
 
-# Build a list of include directories by putting -I in front of each include dir.
+# Build a list of include directories
 function(get_include_directories IncludeDirectories)
     get_directory_property(dirs INCLUDE_DIRECTORIES)
     foreach(dir IN LISTS dirs)
+
+      if (CLR_CMAKE_PLATFORM_ARCH_ARM AND WIN32)
+        list(APPEND INC_DIRECTORIES /I${dir})
+      else()
         list(APPEND INC_DIRECTORIES -I${dir})
+      endif(CLR_CMAKE_PLATFORM_ARCH_ARM AND WIN32)
+
     endforeach()
     set(${IncludeDirectories} ${INC_DIRECTORIES} PARENT_SCOPE)
 endfunction(get_include_directories)
 
+# Build a list of include directories for consumption by the assembler
+function(get_include_directories_asm IncludeDirectories)
+    get_directory_property(dirs INCLUDE_DIRECTORIES)
```

Comment:
Donno why my comment got truncated. 
wouldn't armasm accept -I infront of each include directory instead of -I listof semicolon delimited dir paths

TFS builds actually pass exactly that to armasm i.e. -I infront of each include directory . So we should not be requiring get_include_directories_asm(). get_include_directories() should work for armasm as well.